crates/fuzzing/src/generators/single_inst_module.rs

```diff
@@ -105,205 +105,168 @@ macro_rules! valtype {
     };
 }
 
-macro_rules! binary {
-    ($inst:ident, $rust_ty:tt) => {
-        binary! { $inst, $rust_ty, $rust_ty }
+macro_rules! inst {
+    ($inst:ident, ($($arguments_ty:tt),+) -> $result_ty:tt) => {
+        inst! { $inst, ($($arguments_ty),+) -> $result_ty, |_| true }
     };
-    ($inst:ident, $arguments_ty:tt,  $result_ty:tt) => {
+    ($inst:ident, ($($arguments_ty:tt),+) -> $result_ty:tt, $feature:expr) => {
```

Comment:
I don't off-hand see any reason why there has to be a minimum of one instruction argument. I'd think all the repetitions around `$arguments_ty` in both macro cases could use `*` instead of `+`. But given that all the current instructions do have at least one argument, obviously that's not a necessary change.